packages/core/src/Nodes/FunctionNode.ts

```diff
@@ -28,71 +28,76 @@ export class FunctionNode extends Node {
   }
 }
 
+const alpha = 'abcdefghijklmnop';
+const getAlphabeticalKey = (index: number) => alpha[index];
+
+/** Converts list of sockets specifying value type names to an ordeered list of sockets,
+ */
+const toOrderedSockets = (
+  socketValueTypes: string[] | string | undefined = [],
+  getKey: (index: number) => string
+) => {
+  // const alpha = 'abcdefghijklmnop';
+
+  return [...socketValueTypes].map((socketElement, index) => {
+    return new Socket(socketElement, getKey(index));
+  });
+};
+
 export class FunctionDesc extends NodeDescription {
-  constructor(props: {
+  constructor({
+    in: inputValueTypes,
+    out,
+    name,
+    label,
+    aliases,
+    exec
+  }: {
     name: string;
     label?: string;
     category?: NodeCategory;
     aliases?: string[];
-    in?: { [name: string]: string }[] | string[];
```

Comment:
Until we are calling it with key value pair, lets not expose this as an argument;